### ISSUE TITLE
Do not populate postmeta table if a custom field is empty on post add

### ISSUE THREAD

**Issue by GoogleCodeExporter** [viktorthang/wordpress-custom-content-type-manager#293]
```
What functionality do you want to see?

When adding a custom content type, if a custom field is empty, there shouldn't 
be an entry for it in the postmeta table (currently, the entry is created, but 
with an empty meta_value, which adds needless overhead).

Do you have examples of similar functionality on other sites or in other
plugins? (Please share URLs below)

No.

Please provide any additional information below.

This feature request was born out of this topic - 
http://wordpress.org/support/topic/plugin-custom-content-type-manager-dont-creat
e-database-row-if-field-is-empty
```

Original issue reported on code.google.com by `rosca.vlad` on 19 Jan 2012 at 8:47

**Comment by GoogleCodeExporter** [comment]
```
A global setting could be added to control this behavior.
```

Original comment by `ever...@fireproofsocks.com` on 19 Jan 2012 at 5:37
- Changed state: **Accepted**
- Added labels: **Milestone-Release0.9.6**

**Comment by GoogleCodeExporter** [comment]
```
Implemented in 0.9.5.6. Committed revision 494629.

Any empty string field value will cause the row in postmeta to be deleted.

Actually, I probably need to check for [""] values in repeatable (i.e. multi) 
fields too...
```

Original comment by `ever...@fireproofsocks.com` on 24 Jan 2012 at 2:55
- Changed state: **InProgress**

**Comment by GoogleCodeExporter** [comment]
```
Ok, I've got the code that checks for empty JSON arrays (as might result from 
repeatable fields).

Committed revision 495002.  This will go live in 0.9.5.6.
```

Original comment by `ever...@fireproofsocks.com` on 25 Jan 2012 at 3:57
- Changed state: **Fixed**
